app/src/test/java/org/oppia/android/app/player/state/StateFragmentLocalTest.kt

```diff
@@ -267,6 +268,67 @@ class StateFragmentLocalTest {
     }
   }
 
+  @Test
+  fun testContinueInteractionAnim_openPrototypeExp_checkContinueButtonAnimatesAfter45Seconds() {
+    TestPlatformParameterModule.forceEnableContinueButtonAnimation(true)
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withId(R.id.state_recycler_view)).perform(scrollToViewType(CONTINUE_INTERACTION))
+      onView(withId(R.id.continue_button)).check(matches(Matchers.not(isAnimating())))
+      testCoroutineDispatchers.advanceTimeBy(45000)
+      onView(withId(R.id.continue_button)).check(matches(isAnimating()))
+    }
+  }
+
+  @Test
+  fun testConIntAnim_openProtExp_orientLandscapeAfter30Sec_checkAnimStartsIn45SecAfterOpening() {
+    TestPlatformParameterModule.forceEnableContinueButtonAnimation(true)
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      testCoroutineDispatchers.runCurrent()
+
+      scrollToViewType(CONTINUE_INTERACTION)
+      onView(withId(R.id.continue_button)).check(matches(Matchers.not(isAnimating())))
+      testCoroutineDispatchers.advanceTimeBy(30000)
+      onView(isRoot()).perform(orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      scrollToViewType(CONTINUE_INTERACTION)
+      onView(withId(R.id.continue_button)).check(matches(Matchers.not(isAnimating())))
+      testCoroutineDispatchers.advanceTimeBy(15000)
+      onView(withId(R.id.continue_button)).check(matches(isAnimating()))
```

Comment:
This indicates two different tests--I suggest splitting it up, e.g.:

- Doesn't animate after rotating with a 30 second wait
- Does animate if another 15 seconds is waited

You should also remove/split out the check above. Each test should *only* have checks at the end of the test, not at the middle.

Reply:
done